src/together/resources/finetune.py

```diff
@@ -397,6 +464,15 @@ def download(
             FinetuneDownloadResult: Object containing downloaded model metadata
         """
 
+        if re.match(r"^ft-[\dabcde-]+\:\d+$", id) is not None:
+            if checkpoint_step == -1:
+                checkpoint_step = int(id.split(":")[1])
+                id = id.split(":")[0]
```

Comment:
Not sure if I understand this: if `checkpoint_step` equals -1, looks like we're just parsing the step from the ID, right? This is not the same as downloading the final model

Reply:
Change the default to None